caom2/caom2/caom_util.py

```diff
@@ -116,16 +115,15 @@ def validate_path_component(caller, name, test):
     name : name of the component
     test : component to be tested
 
-    An assertionError is thrown when the the provided test argument
+    An ValueError is thrown when the the provided test argument
     is invalid
     """
 
-    assert (' ' not in test and
-            '/' not in test and
-            '||' not in test and
-            '%' not in test), (
-        caller.__class__.__name__ + ": invalid " + name +
-        ": may not contain space ( ), slash (/), escape (\\), or percent (%)")
+    if ' ' in test or '/' in test or '||' in test or '%' in test:
+        raise \
+            ValueError(caller.__class__.__name__ + ": invalid " + name +
+                       ": may not contain space ( ), slash (/), escape (\\), "
+                       "or percent (%)")
```

Comment:
Not sure how to comment on code that is not changed, so I'm putting this comment here: there are still 'assert' statements in common.py, diff.py, obs_reader_writer.py, observation.py, plane.py and shape.py.

Reply:
Phew. `caom2` is now `assert` free!